### PR TITLE
fix: single quote support #1362

### PR DESCRIPTION
**Pull request by Cerebrovinny**

### CODE REVIEW

File: pkg/utils/string_utils.go [hunk 1]
```diff
@@ -22,26 +22,47 @@
...
+func trimMatchingQuotes(value string) string {
+	if len(value) >= 2 {
+		first := value[0]
+		last := value[len(value)-1]
+		if (first == '\'' || first == '"') && first == last {
+			return value[1 : len(value)-1]
+		}
+	}
+	return value
+}
```
[the remark this answers is not in the document]
Comment (Cerebrovinny) [reply pkg/utils/string_utils.go:89]: @coderabbitai this will be very risky because YAML tags like !terraform.output relies on positional arguments so If someone passes an empty string the trimmed value would be empty, we’d continue, and the slice would collapse to two elements, breaking the call

File: pkg/utils/string_utils.go [hunk 2]
```diff
@@ -22,26 +23,48 @@ func UniqueStrings(input []string) []string {
...
+		if trimmed == "" {
+			continue
 		}
+		filteredParts = append(filteredParts, trimmed)
 	}
 
 	return filteredParts, nil
 }
 
+func trimMatchingQuotes(value string) string {
```
Comment (osterman) [inline pkg/utils/string_utils.go:65]: Please add a function comment that explains how it works and what it's used for.

File: pkg/utils/string_utils_test.go [hunk 3]
```diff
@@ -45,6 +45,27 @@ func TestSplitStringByDelimiter(t *testing.T) {
 			expected:  []string{"foo:", "!env", "FOO"},
 			expectErr: false,
 		},
+		{
+			name:      "Single quoted value with nested double quotes",
+			input:     "core '.security.users[\"github-dependabot\"].access.key.id'",
+			delimiter: ' ',
+			expected:  []string{"core", ".security.users[\"github-dependabot\"].access.key.id"},
+			expectErr: false,
```
Comment (osterman) [inline pkg/utils/string_utils_test.go:57]: Can you maybe generate more tests, especially for edge cases? Such as if somebody nests single quotes inside of a single quote.

File: pkg/utils/string_utils.go (hunk 2 above)
Comment (Cerebrovinny) [reply pkg/utils/string_utils.go:65]: added and clean up also :)

**Comment by osterman** [comment]
@Cerebrovinny also need to update the docs that this is supported

File: website/docs/functions/yaml/include.mdx [hunk 4]
```diff
@@ -262,6 +262,21 @@ On macOS and Linux:
     description: !include '"component description.md"'
 ```
 
+**Using YQ Expressions with Bracket Notation**
+
+When a YQ expression needs double quotes inside bracket notation (for example, `["github-dependabot"]`), wrap the entire `!include` call in single quotes so the parser treats it as one argument. The detailed guidance lives in the [!terraform.output documentation](/functions/yaml/terraform.output#handling-yq-expressions-with-bracket-notation-and-quotes), which applies here as well.
```
Comment (osterman) [inline website/docs/functions/yaml/include.mdx]: This is misleading. Hearing when we say wrap the entire include call, I don't think that A) that invalidates the YAML tag. I think what we mean to say here is to wrap the arguments passed to include. Include itself should never exist inside of strings—because then it's just a string—it's not a YAML tag, which is a feature of YAML.

File: pkg/utils/string_utils_test.go [hunk 5]
```diff
@@ -49,6 +49,41 @@ func TestSplitStringByDelimiter(t *testing.T) {
...
+		{
+			name:      "Single quoted value with escaped single quotes",
+			input:     "core '.security.users[''github-dependabot''].access.key.id'",
+			delimiter: ' ',
+			expected:  []string{"core", ".security.users['github-dependabot'].access.key.id"},
+			expectErr: false,
+		},
+		{
+			name:      "Double quoted value with escaped double quotes",
+			input:     "\"foo\"\"bar\" baz",
```
Comment (aknysh) [inline pkg/utils/string_utils_test.go:70]: @Cerebrovinny the PR looks good to me.
I was looking at `"\"foo\"\"bar\" baz"`, which is difficult to understand.
It means the following:

```
If the parser sees "" (two consecutive double quotes inside a quoted string), according to CSV/Excel-like conventions, a "" inside quotes means a literal " character in the final value.
```

can you please add that comment to the test so it's easier to understand by looking at it?

Comment (aknysh) [reply pkg/utils/string_utils_test.go:70]: i added the comment

**Comment by aknysh** [comment]
@coderabbitai full review
